src/idiomorph.js

```diff
@@ -1063,11 +1074,49 @@ var Idiomorph = (function () {
         function removeNode(tempNode, ctx) {
             removeIdsFromConsideration(ctx, tempNode)
             if (ctx.callbacks.beforeNodeRemoved(tempNode) === false) return;
-
-            tempNode.parentNode?.removeChild(tempNode);
+            if (ctx.pantry) {
+                moveToPantry(tempNode.pantry, ctx);
+            } else {
+                tempNode.parentNode?.removeChild(tempNode);
+            }
             ctx.callbacks.afterNodeRemoved(tempNode);
         }
 
+        function moveToPantry(node, ctx) {
+            if (!node) return;
```

Comment:
rather than a recursive algo, do a query selector on `[id]` to grab all elements that have ids?

might be better for perf

Reply:
Agree, except is it possible to do bottom-up in that way? Maybe that's where the idSets can come into the picture?